api/src/OpenTelemetry/Attributes.hs

```diff
@@ -97,7 +100,7 @@ addAttribute AttributeLimits{..} Attributes{..} k v = case attributeCountLimit o
 
 addAttributes :: ToAttribute a => AttributeLimits -> Attributes -> [(Text, a)] -> Attributes
 -- TODO, this could be done more efficiently
-addAttributes limits = foldl (\attrs' (k, v) -> addAttribute limits attrs' k v)
+addAttributes limits = foldl' (\(!attrs') (!k, !v) -> addAttribute limits attrs' k v)
```

Comment:
The bang on `attrs'` is redundant because we're using `foldl'`:

```suggestion
addAttributes limits = foldl' (\attrs' (!k, !v) -> addAttribute limits attrs' k v)
```